src/lgdo/types/table.py

```diff
@@ -140,8 +140,19 @@ def add_field(
 
         super().add_field(name, obj)
 
+        if not self.size:
+            self.size = len(obj)
+
         # check / update sizes
         if self.size != len(obj):
+            warn(
```

Comment:
I think this should be a `logging.warning()`.

Reply:
`warn()` was also used below in `get_dataframe()`. But we can also replace that one with the logging approach.

Reply:
Yeah I am not sure about the best thing to do in this case see also discussion here:

https://stackoverflow.com/questions/9595009/warnings-warn-vs-logging-warning

You might be right that `warnings.warn()` is better here.